frontend/src/utilities/__tests__/addTypesToK8sListedReources.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { addTypesToK8sListedResources } from '~/utilities/addTypesToK8sListedResources';
+
+const servingRuntimeTemplate = {
+  apiVersion: 'template.openshift.io/v1',
+  items: [
+    {
+      apiVersion: 'serving.kserve.io/v1alpha1',
+      kind: 'ServingRuntime',
+      metadata: {
+        name: 'test-model',
+        annotations: {
+          'openshift.io/display-name': 'New OVMS Server',
+        },
+        labels: {
+          'opendatahub.io/dashboard': 'true',
+        },
+      },
+    },
+  ],
+  metadata: {
+    resourceVersion: '24348645',
+    continue: '',
+  },
+};
+
+const templateWithKind = {
+  apiVersion: 'v1',
+  apiGroup: 'template.openshift.io',
+  kind: 'Template',
+  plural: 'templates',
+};
+
+const templateWithoutKind = {
+  apiVersion: 'v1',
+  apiGroup: 'template.openshift.io',
+  kind: '',
+  plural: 'templates',
+};
+
+describe('addTypesToK8sListedResources', () => {
+  it('should have kind as Template', () => {
+    expect(servingRuntimeTemplate.items[0].kind).not.toBe('Template');
+    expect(
+      addTypesToK8sListedResources(servingRuntimeTemplate, templateWithKind).items[0].kind,
+    ).toBe('Template');
```

Comment:
Test that the function returns a new instance and assigns both `kind` and `apiVersion` for each item.
```suggestion
    const list = addTypesToK8sListedResources(servingRuntimeTemplate, templateWithKind);
    expect(list).not.toBe(servingRuntimeTemplate);
    expect(list.items).toHaveLength(servingRuntimeTemplate.items.length);
    list.items.forEach((i) => {
      expect(i.kind).toBe('Template');
      expect(i.apiVersion).toBe('template.openshift.io/v1');
    });
```